File: justViewer/project_JustViewer/src/main/java/com/kds3393/just/viewer/Image/ImageViewerConfig.java
package com.kds3393.just.viewer.Image;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

import com.kds3393.just.viewer.provider.DBItemData;

public class ImageViewerConfig extends FrameLayout {
	
	public ImageViewerConfig(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public ImageViewerConfig(Context context) {
		super(context);
	}
	
	protected DBItemData mImageData;
	
	protected int mSpacing = 10;
	protected int movePageHThreshold = 0;
	protected int mAnimationDuration = 300;
	
	protected int mPageCount = 0;
	
	public void setImageData(DBItemData data) {
		mImageData = data;
	}
	public DBItemData getImageData() {
		return mImageData;
	}
	public int getIndex() {
		return mImageData.mPageNum;
	}

	public void setPageCount(int count) {
		mPageCount = count;
	}

	public int getPageCount() {
		return mPageCount;
	}
}

File: JustViewer2/app/src/main/java/com/kds3393/just/justviewer2/image/ImageViewerConfig.kt
package com.kds3393.just.justviewer2.image

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout
import com.kds3393.just.justviewer2.data.BookInfo

open class ImageViewerConfig @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : FrameLayout(context,attrs) {
    lateinit var bookInfo: BookInfo
    protected var mSpacing = 10
    protected var movePageHThreshold = 0
    protected var mAnimationDuration = 300
    var pageCount = 0
    fun getPageIndex() : Int {
        return if (this::bookInfo.isInitialized) bookInfo.currentPage else -1
    }
}